nuto/mechanics/solver/Solve.cpp

```diff
@@ -1,23 +1,99 @@
 #include "Solve.h"
 #include "nuto/math/EigenSparseSolve.h"
+#include "nuto/mechanics/dofs/DofVectorConvertEigen.h"
+#include "nuto/mechanics/dofs/DofMatrixSparseConvertEigen.h"
 
-namespace NuTo
+using namespace NuTo;
+
+DofVector<double> NuTo::Solve(const DofMatrixSparse<double>& K, const DofVector<double>& f,
+                              Constraint::Constraints& bcs, std::vector<DofType> dofs, std::string solver)
 {
+    auto K_full = ToEigen(K, dofs);
+    auto f_full = ToEigen(f, dofs);
+
+    DofMatrixSparse<double> C_dof;
+    for (auto rdof : dofs)
+        C_dof(rdof, rdof) = bcs.BuildUnitConstraintMatrix2(rdof, f[rdof].rows());
+
+    for (auto rdof : dofs)
+        for (auto cdof : dofs)
+            if (rdof.Id() != cdof.Id())
+                C_dof(rdof, cdof) = Eigen::SparseMatrix<double>(C_dof(rdof, rdof).rows(), C_dof(cdof, cdof).cols());
 
-GlobalDofVector Solve(GlobalDofMatrixSparse K, GlobalDofVector f, Constraint::Constraints bcs, DofType dof,
-                      int numIndependentDofs, double time, std::string solver)
+    auto C = ToEigen(C_dof, dofs);
+    Eigen::SparseMatrix<double> Kmod = C.transpose() * K_full * C;
+    Eigen::VectorXd fmod = C.transpose() * f_full;
+
+    Eigen::VectorXd u = EigenSparseSolve(Kmod, fmod, solver);
+    u = C * u;
+
+    // TODO: for correct size
+    DofVector<double> result = f;
+    FromEigen(u, f.DofTypes(), &result);
+    return result;
+}
+
+DofVector<double> NuTo::SolveTrialState(const DofMatrixSparse<double>& K, const DofVector<double>& f, double oldTime,
```

Comment:
I am not sure about this method here. `Solve` generally solves the system once. This can be used in for linear solves or nonlinear iterations. 
I understand that we may want to reuse `SolveTrialState` in the future. For now, it is used once in `QuasistaticSolver` and could be implemented differently there. E.g. we do not need to build the full constraint matrix from scratch in every call. I mean, before this PR, the constraint matrix contained mostly zeros and I would not care. Now, it has loads of ones in there. But my concerns are mainly conceptually. That this method is very specific for softening problems. (although it does not hurt in other problems... not sure.)

In any case it contains code duplication. Why not call `Solve(K, K*deltaBrhs, ...)`?

Reply:
This method is not specific for softening problems, since for other linear problems it directly gives the correct result in one iteration (without significant additional effort). 
Regarding the code duplication: There is some code duplication, however doing it the way as you describe above would require to implement the DofMatrix DofVector multiplication, but as stated above, I don't think this is the way to go.